src/Sdk/DTExpressions2/Expressions2/Sdk/Functions/Replace.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+
+namespace GitHub.DistributedTask.Expressions2.Sdk.Functions
+{
+    internal sealed class Replace : Function
+    {
+        protected sealed override Boolean TraceFullyRealized => false;
+
+        protected sealed override Object EvaluateCore(
+            EvaluationContext context,
+            out ResultMemory resultMemory)
+        {
+            resultMemory = null;
+            var value = Parameters[0].Evaluate(context);
+            var characterToReplace = Parameters[1].Evaluate(context);
+            var characterToReplaceWith = Parameters[2].Evaluate(context);
+
+            if (value.IsPrimitive)
+            {
+                value = value.ConvertToString();
```

Comment:
converttostring is safe to always call (even if it's already a string) more sense to change to `var value = Parameters[0].Evaluate(context).ConvertToString()` above;

Reply:
or i guess if any of the values are not a string then could detect and either throw or convert to empty string... not sure which is better.